src/Collection/CollectionTrait.php

```diff
@@ -135,10 +135,20 @@ public function map(callable $c)
      * {@inheritDoc}
      *
      */
-    public function reduce(callable $c, $zero)
+    public function reduce(callable $c, $zero = null)
     {
+        $skip = false;
+        if ($zero === null) {
+            $zero = $this->first();
+            $skip = true;
+        }
+
         $result = $zero;
         foreach ($this->_unwrap() as $k => $value) {
+            if ($skip) {
+                $skip = false;
+                continue;
```

Comment:
That's how underscore js and other libraries work as well. Definitely a case against modifying environment instead of using pure functions. 

I personally think the improvement offsets the drawbacks. If you prefer modifying external vars, better use `each()`